### PR TITLE
feat: arcsinh_transformation and unit tests

### PR DESCRIPTION
**Pull request by fangliu117**
@georgezakinih Please review the arcsinh_transformation function at your convenience. Thanks!

### CODE REVIEW

File: src/spac/transformations.py [hunk 1]
```diff
@@ -496,6 +498,89 @@ def normalize_features(
...
+        # Check for zero values in factor and replace them to avoid division
+        # by zero
+        factor[factor == 0] = 1e-10
+
+    # Apply the arcsinh transformation using the co-factor
+    transformed_data = np.arcsinh(data_to_transform / factor)
+
+    # Check if output_layer already exists and issue a warning if it does
+    if output_layer in adata.layers:
+        warnings.warn(
```
Comment (georgezakinih) [inline src/spac/transformations.py:569]: can we use logging instead?

File: tests/test_transformations/test_arcsinh_transformation.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            [1.50575679, 1.50575679]
+        ])
+        self.assertIn(output_name, transformed_adata.layers)
+        np.testing.assert_array_almost_equal(
+            transformed_adata.layers[output_name], expected_data
+        )
+
+    def test_invalid_percentile_handling(self):
+        with self.assertRaises(ValueError) as context:
+            arcsinh_transformation(self.adata, percentile=-5)
```
Comment (georgezakinih) [inline tests/test_transformations/test_arcsinh_transformation.py:77]: @fangliu117 , please use -0.1 as this is the corner case.

File: tests/test_transformations/test_arcsinh_transformation.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        self.assertEqual(
+            str(context.exception), "Percentile should be between 0 and 100."
+        )
+
+        with self.assertRaises(ValueError) as context:
+            arcsinh_transformation(self.adata, percentile=105)
```
Comment (georgezakinih) [inline tests/test_transformations/test_arcsinh_transformation.py:83]: @fangliu117 , please use 100.1 as this is the corner case.

File: tests/test_transformations/test_arcsinh_transformation.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        expected_data = np.array([
+            [0.88137359, 0.530343],
+            [0.88137359, 1.283796],
+            [1.44363548, 1.480294]
+        ])
+        np.testing.assert_array_almost_equal(
+            transformed_adata.layers['arcsinh'], expected_data
+        )
+
+    def test_arcsinh_transformation_fixed_co_factor(self):
```
Comment (georgezakinih) [inline tests/test_transformations/test_arcsinh_transformation.py:48]: @fangliu117 , would you please add a test of cofactor taking precedence over quantile?

**Review by georgezakinih — CHANGES_REQUESTED** [review]
